app/server/integration/account-resolvers.test.ts

```diff
@@ -0,0 +1,75 @@
+import bcrypt from 'bcryptjs';
+import { getOrCreateServer } from '../src/core/server';
+import { getPrismaClient } from '../src/core/utils';
+import { createMercuriusTestClient } from 'mercurius-integration-testing';
+import * as plugins from '../src/core/plugins';
+import { redisEmitter } from '../src/core/plugins/mercurius';
+import * as jwt from '../src/lib/jwt';
+import faker from '@faker-js/faker';
+import { toGlobalId } from '../src/features/utils';
+
+const server = getOrCreateServer();
+const testClient = createMercuriusTestClient(server);
+const prisma = getPrismaClient(server.log);
+
+const toUserId = toGlobalId('User');
+const userData = {
+    email: faker.internet.email(),
+    firstName: faker.name.firstName(),
+    lastName: faker.name.lastName(),
+    id: faker.datatype.uuid(),
+    password: faker.internet.password(),
+    preferredLang: 'EN',
+};
+
+beforeAll(async () => {
+    plugins.attachCookieTo(server);
+    plugins.attachMercuriusTo(server);
+
+    const encryptedPassword = await bcrypt.hash(userData.password, 10);
```

Comment:
just trying to make sure all the data is being set properly and accurately. The password is always encrypted before being put in the database so why would test cases be any different? I also think it's best if the test user password is encrypted so we that things like login are working properly including the encrypted password validation.

Reply:
I think my main concern here is that it seems like we're re-implementing some of the business logic in the test.  The test setup has to be aware of the specific way in which bcrypt is called (with 10 salt rounds).

I think a better test design may be to spy on `bcrypt.hash` and just ensure that it's called, but not necessarily _how_ it's called.  Although, I'm open to other ideas too.

Reply:
I'm confused on what you mean? We are only hashing the passwords for the expected outputs on methods like register that will hash it so if we didn't hash the expected output the test would fail. Maybe I'm mistaken on what you're asking, you'll have to explain a bit more on how spying would be a replacement for the current hashing that is done.